### PR TITLE
1.0: in_http: Add Authorization auth mechanism document

### PR DESCRIPTION
**Pull request by cosmo0920**
Signed-off-by: Hiroshi Hatake <hatake@calyptia.com>

### CODE REVIEW

File: input/http.md [hunk 1]
```diff
@@ -160,6 +160,20 @@ Responds with an empty GIF image of 1x1 pixel \(rather than an empty string\).
...
 Respond status code with 204. This option will be deprecated at v2 because fluentd v2 will respond 204 as default.
 
+### `authorization_token`
+
+| type | default | version |
+| :--- | :--- | :--- |
+| string | nil\(disabled\) | v1.14.4 |
+
+If you set `Basic <<Base64 encoded string>>` to `authorization_token`, `in_http` verifies `Authorization` HTTP header as token for authentication.
+This authorization_token can handle not only Basic authentication but also other types of authentications.
```
Comment (kenhys) [inline input/http.md]:
```suggestion
This `authorization_token` can handle not only Basic authentication but also other types of authentications.
```

**Comment by cosmo0920** [comment]
Hmm, using GitHub suggestion commit breaks DCO. I'll fix it later.

File: input/http.md [hunk 2]
```diff
@@ -160,6 +160,20 @@ Responds with an empty GIF image of 1x1 pixel \(rather than an empty string\).
...
 
+### `authorization_token`
+
+| type | default | version |
+| :--- | :--- | :--- |
+| string | nil\(disabled\) | v1.14.4 |
+
+If you set `Basic <<Base64 encoded string>>` to `authorization_token`, `in_http` verifies `Authorization` HTTP header as token for authentication.
+This `authorization_token` can handle not only Basic authentication but also other types of authentications.
+`authentication_token` will be interpreted as `<type>` and `<credentials>`:
```
Comment (ashie) [inline input/http.md:171]:
```diff
- authentication_token
+ authorization_token
```

File: input/http.md [hunk 3]
```diff
@@ -160,6 +160,20 @@ Responds with an empty GIF image of 1x1 pixel \(rather than an empty string\).
...
+
+```
+Authorization: <type> <credentials>
+```
+
```
Comment (ashie) [inline input/http.md:176]: Could you describe an example value like this?: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization#basic_authentication

**Review by ashie — CHANGES_REQUESTED** [review]

**Comment by ashie** [comment]
[The patch](https://github.com/fluent/fluentd/pull/3572) is rejected.

**Pull request closed without merging**